src/transformers/portable-text-transformer/portable-text-transformer.ts

```diff
@@ -51,20 +52,27 @@ import {
     ValidElement,
     BlockElement,
     IgnoredElement,
-    TransformFunction
+    TransformFunction,
+    TransformTableCellFunction
 } from "../../utils/index.js"
 
 /**
  * Transforms a parsed tree into an array of Portable Text Blocks.
  *
  * This function takes the parsed tree of a rich text content, flattens it to an array of intermediate
  * Portable Text Objects, and then composes and merges these objects into an array of Portable Text Blocks.
+ * An optional extension function can be provided to apply additional transformations to each Portable Text Object
+ * during the flattening process.
  *
- * @param {IOutputResult} parsedTree - The parsed tree structure representing the rich text content.
+ * @param {IOutputResult} parsedTree The parsed tree structure representing the rich text content.
+ * @param {ExtendPortableTextFunction<PortableTextObject>} [extensionFunction] Optional function to apply
+ *        additional transformations to each Portable Text Object. This function should accept a Portable Text Object
+ *        as its argument and return a transformed Portable Text Object.
  * @returns {PortableTextBlock[]} An array of Portable Text Blocks representing the structured content.
  */
-export const transformToPortableText = (parsedTree: IOutputResult): PortableTextBlock[] => {
-    const flattened = flatten(parsedTree.children);
+export const transformToPortableText = (parsedTree: IOutputResult, extensionFunction?: ExtendPortableTextFunction<PortableTextObject>): PortableTextBlock[] => {
+    const flattened = flatten(parsedTree.children, extensionFunction);
```

Comment:
You can call the `extensionFunction` on the final result of the `flatten` function. It would be easier and you wouldn't have to propagate the function through all the different calls of `flatten`.

Reply:
I was wondering what the better approach is. injecting the `extensionFunction` to the transform method allows you to manipulate object that have not been merged yet, thus avoiding the need to mutate nested objects, you can just pick the marks, spans and whatnot straight away, without complicating the logic. downside is, you can mess up the objects prior to merging, which can result in errors if you change something you shouldn't